### PR TITLE
More folding for aten.gt.int, aten.ne.int and Aten__Getitem__TOp.

### PR DESCRIPTION
**Pull request by cathyzhyi**
- Fold more for aten.gt.int, aten.ne.int and Aten__Getitem__TOp
- Some format cleaning up

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -683,11 +694,15 @@ void Torch::ConstantBoolOp::getAsmResultNames(
...
 
-void Aten__Getitem__TOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
-                                                     MLIRContext *context) {
+void Aten__Getitem__TOp::getCanonicalizationPatterns(
+    RewritePatternSet &patterns, MLIRContext *context) {
   patterns.add(+[](Aten__Getitem__TOp op, PatternRewriter &rewriter) {
     auto torchList = op.getOperand(0);
-    if(!torchList.hasOneUse())
+    // TODO: Use a proper effects interface when more operands taking a list
+    // are implemented
```
Comment (silvasean) [inline lib/Dialect/Torch/IR/TorchOps.cpp:702]: Nit: proper punctuation (`.` at the end of the comment)

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 2]
```diff
@@ -434,6 +437,14 @@ OpFoldResult AtenNeIntOp::fold(ArrayRef<Attribute> operands) {
   // `torch.aten.ne.int %x, %x` -> `false`
   if (getOperand(0) == getOperand(1))
     return getI1IntegerAttr(getContext(), false);
+  auto lhs = operands[0].dyn_cast_or_null<IntegerAttr>();
+  auto rhs = operands[1].dyn_cast_or_null<IntegerAttr>();
+  if (lhs && rhs) {
+    if (lhs.getValue().getSExtValue() != rhs.getValue().getSExtValue())
```
Comment (silvasean) [inline lib/Dialect/Torch/IR/TorchOps.cpp:443]: return getI1IntegerAttr(getContext(), lhs.getValue().getSExtValue() != rhs.getValue().getSExtValue());

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 3]
```diff
@@ -422,6 +423,8 @@ OpFoldResult AtenGtIntOp::fold(ArrayRef<Attribute> operands) {
   if (lhs && rhs) {
     if (lhs.getValue().getSExtValue() > rhs.getValue().getSExtValue())
       return getI1IntegerAttr(getContext(), true);
+    else
```
Comment (silvasean) [inline lib/Dialect/Torch/IR/TorchOps.cpp:426]: return getI1IntegerAttr(getContext(), lhs.getValue().getSExtValue() > rhs.getValue().getSExtValue());

File: test/Dialect/Torch/canonicalize.mlir [hunk 4]
```diff
@@ -30,25 +30,57 @@ func @torch.aten.size$unknown_size(%arg0: !torch.vtensor<[?,3],f32>) -> !torch.l
...
-func @torch.aten.ne.int$same_value(%arg0: !torch.int) -> !torch.bool {
+func @torch.aten.ne.int$same_operand(%arg0: !torch.int) -> !torch.bool {
   %0 = torch.aten.ne.int %arg0, %arg0 : !torch.int, !torch.int -> !torch.bool
   return %0 : !torch.bool
 }
 
+// CHECK-LABEL:   func @torch.aten.ne.int$same_value() -> !torch.bool {
+// CHECK:           %[[VAL_0:.*]] = torch.constant.bool false
+// CHECK:           return %[[VAL_0]] : !torch.bool
+// CHECK:         }
```
Comment (silvasean) [inline test/Dialect/Torch/canonicalize.mlir:65]: Nit: Don't include the final `}` in the checks.

File: test/Dialect/Torch/canonicalize.mlir [hunk 5]
```diff
@@ -30,25 +30,57 @@ func @torch.aten.size$unknown_size(%arg0: !torch.vtensor<[?,3],f32>) -> !torch.l
...
+  %int4 = torch.constant.int 4
+  %int4_0 = torch.constant.int 4
+  %2 = torch.aten.ne.int %int4, %int4_0 : !torch.int, !torch.int -> !torch.bool
+  return %2 : !torch.bool
+}
+
+// CHECK-LABEL:   func @torch.aten.ne.int$different_value() -> !torch.bool {
+// CHECK:           %[[VAL_0:.*]] = torch.constant.bool true
+// CHECK:           return %[[VAL_0]] : !torch.bool
+// CHECK:         }
```
Comment (silvasean) [inline test/Dialect/Torch/canonicalize.mlir:76]: Nit: Don't include the final } in the checks.

**Review by silvasean — APPROVED** [review]
Please address nits and I'll merge for you.

**Comment by cathyzhyi** [comment]
@silvasean Addressed the comments. Thanks for reviewing!

**Pull request merged**